src/imperative.rkt

```diff
@@ -279,7 +280,7 @@
     (define ctx (ctx-reserve-names default-ctx (*reserved-names*)))
 
     (define func-name 
-      (let-values ([(cx fname) (ctx-unique-name ctx (string->symbol name))])
+      (let-values ([(cx fname) (ctx-unique-name ctx (string->symbol (fix-name name)))])
```

Comment:
Why did you add `fix-name` here?

Reply:
Oops. Forgot to remove that. That was part of a change I reverted.